From f31361491416cccc40bb912f1ebbad05d6bae115 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Fri, 10 Sep 2021 10:19:26 +0300
Subject: [PATCH] [SW-47990] gaudi2: init mesh pll

For the I2C slave to be responsive enough while preboot runs from inside
the SRAM, we need to also raise the MESH PLLs.
We raise the frequency according to the half nominal table.

Change-Id: I45fde0594212f560c10a944c305476dc68ef015f
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi2/common/asic.c b/arch/gaudi2/common/asic.c
index aa71b17..d42b3f3 100644
--- a/arch/gaudi2/common/asic.c
+++ b/arch/gaudi2/common/asic.c
@@ -55,6 +55,26 @@
 			     HL_PLL_DIV_PLL
 		},
 	}, {
+		.name = "DCORE0_MESH",
+		.base = mmDCORE0_XBAR_MESH_PLL_CTRL_BASE,
+		.nr = 1, .nf = 34, .od = 1, .nb = 1,	/* 1700 */
+		.div_factor = { 1, 2, 3, 7 },
+		.div_sel = { HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
+		},
+	}, {
+		.name = "DCORE1_MESH",
+		.base = mmDCORE1_XBAR_MESH_PLL_CTRL_BASE,
+		.nr = 1, .nf = 34, .od = 1, .nb = 1,	/* 1700 */
+		.div_factor = { 1, 2, 3, 7 },
+		.div_sel = { HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
+		},
+	}, {
 		.name = "DCORE2_SRAM",
 		.base = mmDCORE2_XBAR_BANK_PLL_CTRL_BASE,
 		.nr = 1, .nf = 64, .od = 2, .nb = 4,
